sensor/tests/resource/pod/pod_test.go

```diff
@@ -77,115 +77,125 @@ func assertDeploymentContainerImages(images ...string) resource.AssertFunc {
 }
 
 func (s *PodHierarchySuite) Test_ContainerSpecOnDeployment() {
-	s.testContext.RunWithResources([]resource.YamlTestFile{
-		NginxDeployment,
-	}, func(t *testing.T, testC *resource.TestContext, objects map[string]k8s.Object) {
-		// wait until pods are created
-		err := wait.For(conditions.New(testC.Resources()).ResourceMatch(objects[NginxDeployment.File], func(object k8s.Object) bool {
-			d := object.(*appsv1.Deployment)
-			return d.Status.AvailableReplicas == 3 && d.Status.ReadyReplicas == 3
-		}), wait.WithTimeout(time.Second*10))
-
-		s.Require().NoError(err)
-
-		testC.LastDeploymentState("nginx-deployment",
-			assertDeploymentContainerImages("docker.io/library/nginx:1.14.2"),
-			"nginx deployment should have a single container with nginx:1.14.2 image")
-
-		messages := testC.GetFakeCentral().GetAllMessages()
-		uniquePodNames := resource.GetUniquePodNamesFromPrefix(messages, "sensor-integration", "nginx-")
-		s.Require().Len(uniquePodNames, 3, "Should have received three different pod events")
-	})
+	s.testContext.NewRun(
+		resource.WithResources([]resource.K8sResourceInfo{
+			NginxDeployment,
+		}),
+		resource.WithTestCase(func(t *testing.T, testC *resource.TestContext, objects map[string]k8s.Object) {
+			// wait until pods are created
```

Comment:
Maybe an idea for another improvement: now that we have these `opts` functions, we could use one `resource.BeforeCase` which provides a function to be executed before the test case and after the resources were created. 

Since there are multiple tests that we wait for `AvailableReplicas == 3` for a resource, this could be a generic function rather than pasted in all test case bodies. 

But this is not for now, just an idea.